src/test/java/com/the/dark/side/crew/fejsbuk/service/impl/PostServiceTest.java

```diff
@@ -2,72 +2,67 @@
 import org.mockito.InjectMocks;
+import org.mockito.Mock;
 import org.mockito.MockitoAnnotations;
-import org.mockito.Spy;
 
-import static org.mockito.Mockito.spy;
-import static org.mockito.Mockito.when;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.*;
 
 public class PostServiceTest {
 
     @InjectMocks
     private PostServiceImpl postService;
 
-    @InjectMocks
-    private PostMapper postMapper;
-
-    @Spy
+    @Mock
     private PostRepository postRepository;
 
-    @Spy
-    private UserRepository userRepository;
+    @Mock
+    private PostMapper postMapper;
 
     @BeforeEach
     void init() {
         MockitoAnnotations.initMocks(this);
     }
 
-//    @Test
-//    void shouldReturnAllPosts() {
-//        // given
-//        long userId = 1L;
-//        PostDto postDto = mock(PostDto.class);
-//        postDto.setUserId(userId);
-//
-////        when(postRepository.findByUserEntityId(userId)).thenReturn(List.of(postEntity));
-//        when(postMapper.toDtos(postRepository.findByUserEntityId(userId))).thenReturn(List.of(postDto));
-//
-//        List<PostDto> result = postService.getAllPosts(userId);
-//        assertTrue(result.);
-//        assertArrayEquals(result, postDto);
-//    }
+    @Test
+    void whenUserFoundThanReturnAllPosts() {
+        long userId = 1L;
+        PostDto postDto = mock(PostDto.class);
+
+        when(postMapper.toDtos(postRepository.findByUserEntityId(userId))).thenReturn(List.of(postDto));
+
+        List<PostDto> result = postService.getAllPosts(userId);
+        assertEquals(1, result.size());
+        assertEquals(postDto, result.get(0));
+    }
+
+    @Test
+    void whenUserNotFoundThanReturnEmptyList() {
+        long userId = 1L;
+
+        when(postMapper.toDtos(postRepository.findByUserEntityId(userId))).thenReturn(List.of());
+
+        List<PostDto> result = postService.getAllPosts(userId);
+        assertEquals(0, result.size());
+    }
 
     @Test
     void shouldAddPost() {
-        // given
-        PostDto postDto = spy(PostDto.class);
-        PostEntity postEntity = spy(PostEntity.class);
-        UserEntity userEntity = spy(UserEntity.class);
-        userEntity.setId(1);
-        postDto.setUserId(1);
-        postEntity.setId(1);
-        postEntity.setUserEntity(userEntity);
-
-        // when
-        when(postRepository.save(postMapper.toEntity(postDto)))
-                .thenReturn(postEntity);
-
-        // then
-//        PostDto post = postMapper.toDto(postService.addPost(postDto));
-//        post.setUserId(1);
-//        verify(postRepository).save(postMapper.toEntity(postDto));
-//        assertEquals(post, postDto);
+        PostDto postDto = mock(PostDto.class);
+        PostEntity postEntity = mock(PostEntity.class);
+
+        when(postMapper.toEntity(postDto)).thenReturn(postEntity);
+        when(postRepository.save(postEntity)).thenReturn(postEntity);
```

Comment:
not necessary, and use verify on .save method

Reply:
when i remove this test fail

Reply:
but it should, u're never using returned refference from this method